src/components/molecules/FilterModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { useRef, useState } from 'react';
+import { CaretLeft } from '@phosphor-icons/react';
+import * as S from '../../styles/molecules/FilterModal';
+import useOnClickOutside from '../../hooks/useOnClickOutside';
+import Image from '../atoms/Image';
+import TagBox from '../atoms/TagBox';
+
+type FilterModalProps = {
+  setModalOpen: (value: boolean) => void;
+};
+
+const dogSize = [
+  { id: 1, name: '소형견' },
+  { id: 2, name: '중형견' },
+  { id: 3, name: '대형견' },
+];
+
+const dogBreed = [
+  { id: 1, name: '리트리버' },
+  { id: 2, name: '닥스훈트' },
+  { id: 3, name: '말티즈' },
+  { id: 4, name: '페키니즈' },
+  { id: 5, name: '푸들' },
+  { id: 6, name: '슈나이저' },
+  { id: 7, name: '보더콜리' },
+  { id: 8, name: '비글' },
+  { id: 9, name: '비숑' },
+  { id: 10, name: '사모예드' },
+  { id: 11, name: '보스턴 테리어' },
+  { id: 12, name: '시츄' },
+  { id: 13, name: '셰틀랜드 쉽독' },
+  { id: 14, name: '시바견' },
+  { id: 15, name: '시베리안 허스키' },
+  { id: 16, name: '웰시 코기' },
+  { id: 17, name: '진돗개' },
+  { id: 18, name: '치와와' },
+  { id: 19, name: '믹스견' },
+  { id: 20, name: '요크셔 테리어' },
+  { id: 21, name: '코커 스패니얼' },
+  { id: 22, name: '이탈리안 그레이하운드' },
+  { id: 23, name: '파피용' },
+  { id: 24, name: '퍼그' },
+  { id: 25, name: '포메라니안' },
+  { id: 26, name: '풍산개' },
+  { id: 27, name: '불독' },
+  { id: 29, name: '미니어처 핀셔' },
+  { id: 30, name: '기타' },
+];
+
+// 검색 창의 필터 아이콘 클릭 시 나타나는 모달
+const FilterModal = ({ setModalOpen }: FilterModalProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const selectedFilter = useState([]);
+
+  // 모달 바깥 클릭 시 모달 닫기
+  useOnClickOutside(ref, () => setModalOpen(false));
+
+  const handleFilterAdap = () => {
+    // 서버 api 요청
+    setModalOpen(false);
+  };
+
+  return (
+    <S.Container role="presentation">
+      <S.ModalWrapper ref={ref}>
+        <S.Modal>
+          <S.ModalHeader>
+            <CaretLeft
+              size={20}
+              onClick={() => setModalOpen(false)}
+              className="modal__header-icon"
+            />
+            &nbsp;필터
```

Comment:
```suggestion
            {' '}필터
```
이렇게 표기하는게 좀더 깔끔할 것 같기는 한데요,
다만 지금과 같은 경우는 띄어쓰기 그 자체가 필요한게 아니라, 좌측 caret과의 여백이 필요한 경우인 것 같습니다.
이런 부분은 css styling으로 처리하는게 적합합니다.